paimon-spark/paimon-spark-ut/src/test/scala/org/apache/paimon/spark/sql/MergeIntoTableTestBase.scala

```diff
@@ -48,6 +48,28 @@ abstract class MergeIntoTableTestBase extends PaimonSparkTestBase with PaimonTab
     }
   }
 
+  test(s"Paimon MergeInto: two paimon table") {
```

Comment:
Sorry. I can't understand the effect of this PR by this test.

Reply:
This would merge from one paimon table to another, which will have two paimon source join and can cover the ambiguous __paimon_file_path case

Reply:
@Aitozi I put this case into the latest master branch, and it seems to execute successfully, can you check it again

Reply:
It still fails 
<img width="1779" alt="image" src="https://github.com/user-attachments/assets/c69c5247-389b-48d0-b84b-0975ce8d1302" />

Reply:
My mistake, append table will fail